_posts/2018-03-15-integrating-redux-and-socketio.markdown

```diff
@@ -0,0 +1,118 @@
+---
+layout: post_longform
+title: "Integrating Redux and Socket.io"
+date: 2018-03-15
+categories: [Technology, positron, javascript]
+author: [luc]
+---
+
+On the engineering team at Artsy, we've built a CMS for both internal and external editors to write and publish articles. We have a team of (x) in-house editors creating new content on a daily basis. As multiple people would use the app simultaneously, something became apparent. Editors would unintentionally go and override each other’s work because there was no way to tell if another editor was currently editing an article. As a workaround, our editors are forced to edit drafts in another editor such as google docs andonly copy their work over once ready. This makes for a lackluster collaborative experience.
+
+So we decided to implement locking mechanism that would make our editors more confident in our CMS by insuring only one editor can edit an article at any given time. I was tasked with coming up with an elegant technical solution for this feature. Here's the approach I took.
+
+<!-- more -->
+
+In order to fix that problem, we decided to build an article locking mechanism so only one of the editors on the team can start editing at once. One of the requirements for this new feature was for our editors to see when someone else started editing without refreshing the page. In order to fulfill this, we needed to implement a system to push events from the server to clients.
+
+Based on the requirements presented, I looked at potential solutions for this. Right away, the HTML5 WebSocket API seemed to be the perfect solution to keep all clients synced in realtime, however a few issues arose. For one, many proxies and firewalls block WebSocket connections so it's not always an available option for clients. That's where socket.io comes in.
```

Comment:
I know you link to it at the end of the post, which is great, but "socket.io" could be a link to https://socket.io here as well.